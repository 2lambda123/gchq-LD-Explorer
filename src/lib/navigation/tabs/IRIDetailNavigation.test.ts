/* (c) Crown Copyright GCHQ */

import type { RoutedTabView } from './types';
import createTabDetail from './IRIDetailNavigation';

describe('IRIDetailNavigation', () => {
	let tabDetail: RoutedTabView<unknown>;
	const iri = 'http://www.example.com/';

	beforeEach(() => {
		tabDetail = createTabDetail(iri);
	});

	it('creates the correct config', () => {
		expect(tabDetail).toMatchSnapshot();
	});

	it('has the heading of whatever IRI is passed in', () => {
		const exampleIri = 'http://www.foobar.com/';
		expect(createTabDetail(exampleIri).heading).toEqual(exampleIri);
	});

	it('has the right type', () => {
		expect(typeof tabDetail).toBe('object');
	});
});
