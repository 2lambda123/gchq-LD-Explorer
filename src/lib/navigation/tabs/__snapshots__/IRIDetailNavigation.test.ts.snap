// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IRIDetailNavigation > creates the correct config 1`] = `
{
  "heading": "http://www.example.com/",
  "navigationLabel": "IRI",
  "subheading": "Detail for a specific IRI",
  "tabs": [
    {
      "description": "Results of running a DESCRIBE query for this IRI. Note that only unique values are shown in the interface, even if that value appears across multiple data sources.",
      "href": "/explore/iris/detail?iri=http%3A%2F%2Fwww.example.com%2F",
      "tabIndex": 0,
      "title": "DESCRIBE query",
    },
    {
      "description": "Any appearances of this IRI within any triple (as either the subject, the predicate or the object).",
      "href": "/explore/iris/detail/appearances?iri=http%3A%2F%2Fwww.example.com%2F",
      "tabIndex": 1,
      "title": "Appearances",
    },
    {
      "description": "Any resources identified to be instances of this resource.",
      "href": "/explore/iris/detail/instances?iri=http%3A%2F%2Fwww.example.com%2F",
      "tabIndex": 2,
      "title": "Instances",
    },
    {
      "description": "Any resources identified to be subclasses of this resource.",
      "href": "/explore/iris/detail/subclasses?iri=http%3A%2F%2Fwww.example.com%2F",
      "tabIndex": 3,
      "title": "Subclasses",
    },
    {
      "description": "Any resources identified to be superclasses of this resource (ancestors).",
      "href": "/explore/iris/detail/superclasses?iri=http%3A%2F%2Fwww.example.com%2F",
      "tabIndex": 4,
      "title": "Superclasses",
    },
  ],
}
`;
